src/imarker_robot_state.cpp

```diff
@@ -108,39 +114,40 @@ bool isIKStateValid(const planning_scene::PlanningScene* planning_scene, bool ve
 
 namespace moveit_visual_tools
 {
-IMarkerRobotState::IMarkerRobotState(planning_scene_monitor::PlanningSceneMonitorPtr psm,
-                                     const std::string& imarker_name, std::vector<ArmData> arm_datas,
-                                     rviz_visual_tools::colors color, const std::string& package_path)
+IMarkerRobotState::IMarkerRobotState(rclcpp::Node::SharedPtr node, planning_scene_monitor::PlanningSceneMonitorPtr psm, const std::string& imarker_name,
+                    std::vector<ArmData> arm_datas, rviz_visual_tools::Colors color, const std::string& package_path)
   : name_(imarker_name)
-  , nh_("~")
   , arm_datas_(std::move(arm_datas))
   , psm_(std::move(psm))
   , color_(color)
   , package_path_(package_path)
 {
   // Load Visual tools with respect to Eigen memory alignment
+  // ! Add node here
   visual_tools_ = std::allocate_shared<moveit_visual_tools::MoveItVisualTools>(
-      Eigen::aligned_allocator<moveit_visual_tools::MoveItVisualTools>(), psm_->getRobotModel()->getModelFrame(),
-      nh_.getNamespace() + "/" + imarker_name, psm_);
+      Eigen::aligned_allocator<moveit_visual_tools::MoveItVisualTools>(), node, psm_->getRobotModel()->getModelFrame(), "/" + imarker_name, psm_);
```

Comment:
Topics starting with a `/` are global. So if we want to have them in the node's namespace (I think we do), then the slash should be dropped.